libvast/src/arrow_table_slice.cpp

```diff
@@ -702,7 +704,10 @@ value_at([[maybe_unused]] const Type& type,
         value_type at(size_type i) const override {
           VAST_ASSERT(!key_array->IsNull(value_offset + i));
           if (item_array->IsNull(value_offset + i))
-            return {};
+            return {
+              value_at(key_type, *key_array, value_offset + i),
+              {},
+            };
```

Comment:
@dominiklohmann the previous implementation created an empty record entry when the value was absent, instead of a record entry with a null value only. I came across this while writing a test for the conversion itself, and verified that this new behavior is identical to the legacy table slice behavior, so I assume it's "more correct".

Reply:
That's definitely a bug fix, thanks for catching this!